### PR TITLE
Sparse field lists and specified related resources

### PR DESCRIPTION
**Pull request by 21echoes**
Make it clear how to use JSON:API to customize the data you receive from our API

### CODE REVIEW

File: lib/patreon/schemas/campaign.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
+require_relative '../../enum/enum'
```
Comment (joshbuddy) [inline lib/patreon/schemas/campaign.rb]: I'd include enum in `lib/patreon.rb` and remove all the relative includes.

Comment (joshbuddy) [inline lib/patreon/schemas/campaign.rb]: (same with `lib/jsonapi/url_util.rb`)

**Comment by joshbuddy** [comment]
The most objectionable thing is polluting the namespace outside of `Patreon`. I'd put `Enum` and `JSONAPI` both under the `Patreon` namespace.

File: lib/patreon/schemas/campaign.rb (hunk 1 above)
Comment (21echoes) [inline lib/patreon/schemas/campaign.rb]: the idea was that those are internal utils, not meant for consumption by clients of this library.. would you still recommend they get put on the public `.rb`?

**Pull request merged**